rasa/core/policies/policy.py

```diff
@@ -354,49 +379,50 @@ def _metadata_filename(cls) -> Text:
         """
         pass
 
-    def persist(self, path: Union[Text, Path]) -> None:
-        """Persists the policy to storage.
+    def persist(self) -> None:
+        """Persists the policy to storage."""
+        with self._model_storage.write_to(self._resource) as path:
+            # not all policies have a featurizer
+            if self.featurizer is not None:
+                self.featurizer.persist(path)
 
-        Args:
-            path: Path to persist policy to.
-        """
-        # not all policies have a featurizer
-        if self.featurizer is not None:
-            self.featurizer.persist(path)
+            file = Path(path) / self._metadata_filename()
 
-        file = Path(path) / self._metadata_filename()
-
-        rasa.shared.utils.io.create_directory_for_file(file)
-        rasa.shared.utils.io.dump_obj_as_json_to_file(file, self._metadata())
+            rasa.shared.utils.io.create_directory_for_file(file)
+            rasa.shared.utils.io.dump_obj_as_json_to_file(file, self._metadata())
```

Comment:
Did we find any reason why metadata should be treated differently from config?

Reply:
yeah, some policies use this to store additional stuff, e.g. the [`RulePolicy` ](https://github.com/RasaHQ/rasa/blob/f78a6a17d5146248500630a2397ee099a24e0f5c/rasa/core/policies/rule_policy.py#L1212) or [MemoizationPolicy](https://github.com/RasaHQ/rasa/blob/107caacd1a25fafac4232e9916875eb9c66bbf9f/rasa/core/policies/memoization.py#L245). 

There is definitely a huge overlap with the config. We should consider this when migrating these policies (doesn't matter for `TEDPolicy`)

Reply:
Yep we can probably split it between config and normal persisted data.